src/libData/AccountData/Account.cpp

```diff
@@ -126,36 +203,112 @@ bool Account::IncreaseNonce()
     return true;
 }
 
-void Account::SetStorage(string _k, string _mutable, string _type, string _v)
+void Account::SetStorageRoot(const h256& root)
+{
+    if (!isContract())
+        return;
+    m_storageRoot = root;
+    if (m_storageRoot != h256())
+    {
+        m_storage.setRoot(m_storageRoot);
+        m_prevRoot = m_storageRoot;
+    }
+}
+
+void Account::SetStorage(string _k, string _type, string _v, bool _mutable)
 {
-    RLPStream rlpStream(3);
-    rlpStream << _mutable << _type << _v;
-    m_storage.insert(
-        bytesConstRef(DataConversion::HexStrToUint8Vec(_k).data(), _k.size()),
-        rlpStream.out());
+    if (!isContract())
+        return;
+    RLPStream rlpStream(4);
```

Comment:
magic number

Reply:
Yes, it's indeed a magic number but actually it refers to the number of elements we will attempt to get via `<<` from `rlpStream`. Not sure if there's a proper way to mitigate this.